test/integration/main_test.go

```diff
@@ -334,33 +347,39 @@ deletion_timeline:
 
 // testGetFirstPmapEventWithRetries queries the DB and get the matched PmapEvent.
 // If no matched PmapEvent was found, the query will be retried with the specified retry inputs.
-func testGetFirstPmapEventWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *config) *v1alpha1.PmapEvent {
+func testGetFirstPmapEventWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *config) (*v1alpha1.PmapEvent, string) {
 	tb.Helper()
 
 	b := retry.NewConstant(cfg.QueryRetryWaitDuration)
-	var pmapEvents []*v1alpha1.PmapEvent
+	var pmapEvent v1alpha1.PmapEvent
+	var attributes BQAttributes
 	if err := retry.Do(ctx, retry.WithMaxRetries(cfg.QueryRetryLimit, b), func(ctx context.Context) error {
-		results, err := queryPmapEvents(ctx, bqQuery)
+		results, err := querysBQRows(ctx, bqQuery)
 		if err != nil {
 			tb.Logf("failed to query pmap events: %v", err)
 			return err
 		}
 
 		// Early exit retry if queried pmap event already found.
 		if len(results) > 0 {
-			pmapEvents = results
+			if err := protojson.Unmarshal([]byte(results[0].Data), &pmapEvent); err != nil {
```

Comment:
I'd suggest moving these two unmarhsal out of this function and write it outside